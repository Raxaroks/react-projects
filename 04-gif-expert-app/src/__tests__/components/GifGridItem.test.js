import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';
 

describe('Pruebas en el componente <GifGridItem />', () => {

    const title = 'Some title';
    const url = 'https://somelink.com/isnotreal';
    const componentWrapper = shallow( <GifGridItem title={title} url={url} /> );

    test('Debe de renderizar y mostrar el componente correctamente.', () => {        
        expect(componentWrapper).toMatchSnapshot();
    })

    test('Debe de tener un párrafo con el título.', () => {
        const p = componentWrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    })

    test('Debe de tener la imagen al url y alt de los props', () => {
        const img = componentWrapper.find('img');
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    })

    test('Debe de tener la clase animada: animate__pulse', () => {
        const div = componentWrapper.find('div');
        const clase = div.prop('className');

        expect( clase.includes('animate__pulse') ).toBe(true);
    })
        
})
