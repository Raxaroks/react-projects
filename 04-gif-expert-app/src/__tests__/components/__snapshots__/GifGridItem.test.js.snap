// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGridItem /> Debe de renderizar y mostrar el componente correctamente. 1`] = `
<div
  className="card animate__animated animate__pulse"
>
  <img
    alt="Some title"
    src="https://somelink.com/isnotreal"
  />
  <p>
    Some title
  </p>
</div>
`;
